include/alpaka/rand/RandPhilox.hpp

```diff
@@ -0,0 +1,232 @@
+            template<typename TScalar>
+            using ResultContainer = typename EngineVariant::template ResultContainer<TScalar>;
+
+            using ResultInt = std::uint32_t;
+            using ResultVec = decltype(engineVariant());
+            ALPAKA_FN_HOST_ACC constexpr ResultInt min()
+            {
+                return 0;
+            }
+            ALPAKA_FN_HOST_ACC constexpr ResultInt max()
+            {
+                return std::numeric_limits<ResultInt>::max();
+            }
+            ALPAKA_FN_HOST_ACC ResultVec operator()()
+            {
+                return engineVariant();
+            }
+            // TODO: https://en.cppreference.com/w/cpp/named_req/RandomNumberEngine
+        };
+
+
+        // The following exists because you "cannot call __device__ function from a __host__ __device__ function"
+        // directly, but wrapping that call in a struct is just fine.
+        template<typename TEngine>
+        struct EngineCallHostAccProxy
+        {
+            ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> decltype(engine())
+            {
+                return engine();
+            }
+        };
+
+        /// TEMP: Distributions to be decided on later. The generator should be compatible with STL as of now.
+        template<typename TResult, typename TSFINAE = void>
+        class UniformReal
+        {
+        public:
+            UniformReal() = delete;
+        };
+
+        /// TEMP: Distributions to be decided on later. The generator should be compatible with STL as of now.
+        template<typename TResult>
+        class UniformReal<TResult, std::enable_if_t<meta::IsArrayOrVector<TResult>::value>>
+            : public concepts::Implements<ConceptRand, UniformReal<TResult>>
+        {
+            using T = typename TResult::value_type;
+
+        public:
+            ALPAKA_FN_HOST_ACC UniformReal() : UniformReal(0, 1)
+            {
+            }
+
+            ALPAKA_FN_HOST_ACC UniformReal(T min, T max) : _min(min), _max(max), _range(_max - _min)
+            {
+            }
+
+            template<typename TEngine>
+            ALPAKA_FN_HOST_ACC TResult operator()(TEngine& engine)
+            {
+                // FIXME: quick hack for 4-vectors, to be changed after API decision.
+                auto result = engine();
+                T scale = static_cast<T>(1) / engine.max() / _range;
+                TResult ret{
+                    static_cast<T>(result[0]) * scale + _min,
+                    static_cast<T>(result[1]) * scale + _min,
+                    static_cast<T>(result[2]) * scale + _min,
+                    static_cast<T>(result[3]) * scale + _min};
+                return ret;
+            }
+
+        private:
+            const T _min;
+            const T _max;
+            const T _range;
+        };
+
+        /// TEMP: Distributions to be decided on later. The generator should be compatible with STL as of now.
```

Comment:
So is this finished now? Or work in progress? If it is the latter, you could remove it from the PR. Or document what is missing or needs be changed later. If the work is finished, then please remove the comment.